src/providers/cognito.js

```diff
@@ -0,0 +1,11 @@
+class SalteAuthCognitoProvider {
+  static deauthorizeUrl(config) {
+    const gateway = config.gateway.replace('/oauth2', '');
```

Comment:
I think we should only accept the base URL as a configuration parameter and then assume that things like "oauth2" have to be appended at the specific provider level.  I realize that this will require both the AuthO and WSO2 providers to append "oauth2" but I still think that would be better than having providers like Cognito have to remove them.

Reply:
Every other provider already expects the `gateway` to be the base url.

Cognito is the exception since its `/authorize` endpoint is hidden under a `/oauth2` path.

As a compromise, I think we should allow `providers` to override the `authorizeUrl` function.

If a provider doesn't implement it, then we'll just append `/authorize` to the `gateway` url.